### PR TITLE
docs revamp: synchronisation planning

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

This adds planning documentation for the org and team synchronization.

**Special notes for your reviewer:**

In order to run the documentation locally using
```
$ cd docs
$ make docs
```
After the docker image is running, the service should be available at http://localhost:3002

@IevaVasiljeva 🙏 Please, can you help me corroborate the information is accurate? I want to make sure I understand the organization sync perfectly.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:169]:
```suggestion
Team sync lets you set up synchronization between your auth providers teams and teams in Grafana. This enables LDAP, OAuth, or SAML users who are members of certain IdP teams or groups to automatically be added or removed as members of certain teams in Grafana.
```

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:171]: Team sync only synchronizes user's teams upon login, so technically it's not "actively querying". You could maybe say something like
```suggestion
Grafana will keep users in sync by querying for any IdP team or group changes upon user login, and adding or removing the user from teams accordingly.
```

Comment (IevaVasiljeva) [reply docs/sources/auth/planning/index.md:171]: I also removed the last two sentences, as they are not really related to team sync. I don't think we automatically remove the users from Grafana if they've been removed from the IdP (but I need to double-check this). And I'd probably mention the labelling under `Internal vs external users` section higher above.

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:175]: Ah, I see that you've added a note about the frequency of sync here. It probably doesn't hurt to mention it twice, as it's something that users are often confused about.

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:181]: Technically, role sync and organization sync are not quite the same. What you describe here sounds like role sync. Org sync is only supported for SAML and LDAP, and also allows mapping users to different organizations (it's not supported in Cloud, as Cloud only has one org).

Here's a useful table with different sync (and other) features and info on what is available for what providers: https://docs.google.com/spreadsheets/d/1ISzqi0mcCs_KRGyvvKd-elMOj2S5EGjOXcUsw_eyxSs/edit#gid=0

Assuming that this section will be about role sync, some points here:
* we should clarify that currently we only support mapping to the basic roles (Viewer, Editor, Admin and Grafana Server Admin);
* similarly as for team sync, the sync happens whenever the user logs in (not just upon the first login);
* it might be worth mentioning that by default we prevent manual role changes for users whose roles are synced by role sync, but there is a feature to allow it (it's what Eric has been working on).

**Review by IevaVasiljeva — APPROVED** [review]
Approving not to block the work, but also adding some suggestions.

I think we should create a new section for role sync, so that we can clearly describe the difference between role sync and org sync.

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [inline docs/sources/auth/planning/index.md:177]: I still feel like this description is a mix of role sync and organization sync.

Here's an example of the current docs for org sync: https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/saml/#configure-organization-mapping

So I'd probably first have a section that describes role sync, and then in this section say something like:

> Organization sync allows mapping users from IdP groups to Grafana organizations. It works similarly as role sync, but in addition allows specifying Grafana organization that a user who belongs to a specific IdP group should be added to. This feature can only be used in self-hosted Grafana instances, as Cloud Grafana instances are limited to one organization.

> Organization sync is currently only supported for SAML and LDAP.

File: docs/sources/auth/planning/index.md
Comment (chri2547) [inline docs/sources/auth/planning/index.md:165]:
```suggestion
When connecting Grafana to an Identity Provider, it's important to think beyond just the initial authentication setup. You should also think about synchronizing user bases and roles. Doing so will enable users within a group to share the same configuration, so you won't have to set individual permissions for each user.
```

File: docs/sources/auth/planning/index.md
Comment (chri2547) [inline docs/sources/auth/planning/index.md:171]:
```suggestion
Team sync is a feature that allows you to synchronize teams or groups from your authentication provider with teams in Grafana. This means that users who are part of specific teams or groups in LDAP, OAuth, or SAML will be automatically added or removed as members of corresponding teams in Grafana. Whenever a user logs in, Grafana will check for any changes in the teams or groups of the authentication provider and update the user's teams in Grafana accordingly. This makes it easy to manage user permissions across multiple systems.
```

File: docs/sources/auth/planning/index.md
Comment (chri2547) [inline docs/sources/auth/planning/index.md:175]:
```suggestion
> **Note:** Currently, team synchronization occurs only when a user logs in. However, if you are using LDAP, it is possible to enable active background synchronization, which was added to Grafana 6.3. This allows for the continuous synchronization of teams.
```

File: docs/sources/auth/planning/index.md
Comment (chri2547) [inline docs/sources/auth/planning/index.md:177]: @linoman , I'm not sure if I should review this section yet. Did you make adjustments based on @IevaVasiljeva feedback?

**Review by chri2547 — COMMENTED** [review]
Some suggestions and I'm not sure if you are ready for me to review the org sync section yet.

File: docs/sources/auth/planning/index.md
Comment (IevaVasiljeva) [reply docs/sources/auth/planning/index.md:165]: What is a user base?

**Comment by linoman** [comment]
@chri2547 I addressed all of the comments. I'm merging this for now to the base branch, not the main branch. If there are any changes, we can work on them before merging this with the main branch.

**Pull request merged**
